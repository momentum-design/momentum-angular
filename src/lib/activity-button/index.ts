export * from './activity-button.module';
export * from './activity-button.component';
