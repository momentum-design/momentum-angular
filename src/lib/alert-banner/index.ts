export * from './alert-banner.module';
export * from './alert-banner.component';
