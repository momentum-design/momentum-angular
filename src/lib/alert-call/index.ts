export * from './alert-call.module';
export * from './alert-call.component';
export * from './alert-call-config';
export * from './alert-call-container.component';
export * from './alert-call.service';
export * from './device-list-call.component';
