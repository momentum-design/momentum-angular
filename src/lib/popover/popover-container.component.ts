import { Component, Input, ChangeDetectionStrategy, TemplateRef, ElementRef} from '@angular/core';
import { animate, style, transition, trigger } from '@angular/animations';

@Component({
  selector: 'cui-popover-container',
  styles: [],
  template: `
  <div class="cui-popover cui-event-overlay cui-event-overlay--arrow">
    <div class="cui-event-overlay__arrow" ></div>
    <div  class="cui-event-overlay__children">
        <cui-popover text={{this.text}} [popoverTemplate]="popoverTemplate"></cui-popover>
    </div>
  </div>
  `
})
export class PopoverContainerComponent {
  /** @prop Sets text in the popover */
  @Input() text: string;
   /** @prop Sets a component in the popover */
  @Input() popoverTemplate?: TemplateRef<any>;

  private _direction: string = null;
  /** @prop Sets direction of the popover | null */
  @Input()
  set direction(direction: string) {
    this.elementRef.nativeElement.children[0].classList.add(`cui-event-overlay--${direction}`);
    this._direction = direction;
  }

  private _showArrow: string;
  /** @prop Sets whether or not to show the arrow on the popover  */
  @Input()
  set showArrow(showArrow: string) {
    if ( showArrow === 'false' ) {
      this.elementRef.nativeElement.children[0].classList.remove(
        `cui-event-overlay--arrow`
      );
      this.elementRef.nativeElement.children[0].children[0].classList.remove(
        `cui-event-overlay__arrow`
      );
    }
    this._showArrow = showArrow;
  }


    constructor(private elementRef: ElementRef) {}
}
