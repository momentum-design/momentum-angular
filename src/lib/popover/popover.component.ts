import { Component, Input, TemplateRef, OnInit, ViewContainerRef} from '@angular/core';


@Component({
  selector: 'md-popover',
  styles: [],
  template: `
  <span>{{ text }}</span>
  <ng-content></ng-content>
  `,
})
export class PopoverComponent implements OnInit {

  @Input() text: string;
  @Input() popoverTemplate: TemplateRef<any>;

  constructor( private _vcr: ViewContainerRef ) {}

  ngOnInit() {
      if ( this.popoverTemplate ) {
          this._vcr.createEmbeddedView(this.popoverTemplate);

      }
  }
}
