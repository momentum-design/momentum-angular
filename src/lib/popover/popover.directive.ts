import { Directive, ElementRef,  Input, TemplateRef } from '@angular/core';
import { Overlay, OverlayPositionBuilder,  ScrollStrategyOptions } from '@angular/cdk/overlay';

import { TooltipDirective } from '../tooltip/tooltip.directive';

@Directive({ selector: '[mdPopover]' })
export class PopoverDirective extends TooltipDirective {

  /** @prop Sets content in the popover can be a string or a template */
  @Input('mdPopover') content: string | TemplateRef<any>;

  /** @prop shows the arrow or not */
  @Input() showArrow: boolean  = true;

  constructor(
    overlay: Overlay,
    overlayPositionBuilder: OverlayPositionBuilder,
    _sso: ScrollStrategyOptions,
    elementRef: ElementRef
    ) {
        super(
          overlay,
          overlayPositionBuilder,
          _sso,
          elementRef
          );
      }

  public show() {
    super.show();
    this.tooltipRef.instance.isTooltip = false;
    this.tooltipRef.instance.classList['md-popover'] = true;
    this.tooltipRef.instance.showArrow = this.showArrow;
  }

  public close() {
    super.close();
  }
}
