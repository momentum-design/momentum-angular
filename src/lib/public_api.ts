/*
 * Public API Surface of collab-ui-angular
 */

export * from './alert';
export * from './alert-banner';
export * from './avatar';
export * from './badge';
export * from './composite-avatar';
export * from './button';
export * from './checkbox';
export * from './icon';
export * from './input-error';
export * from './input-helper';
export * from './input';
export * from './label';
export * from './list-item-section';
export * from './list-item';
export * from './list';
export * from './loading';
export * from './modal-body';
export * from './modal-footer';
export * from './modal-header';
export * from './modal';
export * from './radio';
export * from './spinner';
export * from './tabs';
export * from './toggle-switch';
