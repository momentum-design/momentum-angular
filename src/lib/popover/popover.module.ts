import { NgModule, CUSTOM_ELEMENTS_SCHEMA, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PopoverDirective } from './popover.directive';


@NgModule({
  declarations: [ PopoverDirective ],
  imports: [
    CommonModule
  ],
  exports: [ PopoverDirective ],
  schemas: [ CUSTOM_ELEMENTS_SCHEMA],
})
export class PopoverModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: PopoverModule,
      providers: [],
    };
  }
 }
