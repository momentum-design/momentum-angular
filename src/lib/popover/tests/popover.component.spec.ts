import { async, ComponentFixture, TestBed, inject } from '@angular/core/testing';

import { Component, DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';
import { BadgeModule } from 'src/lib/public_api';
import { TooltipModule } from 'src/lib/public_api';
import { PopoverModule } from '../popover.module';
import { OverlayContainer, Overlay } from '@angular/cdk/overlay';

@Component({
  template: `<md-badge mdPopover="Hello World!">Popover with text only</md-badge>
  <button md-button mdPopover="my button" direction="bottom" showArrow="false" aria-label="myAriaLabel" (click)="onClick()">
        Test Me
      </button>`
})
class TestPopoverComponent {}

describe('Popover Test', () => {
  let component: TestPopoverComponent;
  let fixture: ComponentFixture<TestPopoverComponent>;
  let testEl: DebugElement;
  let overlayContainer: OverlayContainer;
  let overlayContainerEl: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ TestPopoverComponent ],
      imports: [BadgeModule, TooltipModule, PopoverModule],
      providers: [Overlay]
    })
    .compileComponents();
  }));

  beforeEach(inject([OverlayContainer], (container: OverlayContainer) => {
    overlayContainer = container;
    overlayContainerEl = container.getContainerElement();
    fixture = TestBed.createComponent(TestPopoverComponent);
    component = fixture.componentInstance;
    testEl = fixture.debugElement.query(By.css('md-badge'));
    testEl.triggerEventHandler('click', null);
    fixture.detectChanges();
  }));

  it('should match snapshot', () => {
    expect(overlayContainerEl).toMatchSnapshot();
  });

  it('popover should appear when click over badge', () => {

    const popoverEl = overlayContainerEl.querySelector('md-tooltip');
    expect(popoverEl).not.toBeNull();
  });

  it('default popover should have the the default classes', () => {

    const divEl = overlayContainerEl.querySelectorAll('div');
    expect(divEl[2].className).toContain('md-event-overlay md-event-overlay--top md-popover md-event-overlay--arrow');
  });
});
