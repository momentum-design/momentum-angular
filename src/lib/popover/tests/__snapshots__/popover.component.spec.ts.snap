// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Test should match snapshot 1`] = `
<div
  class="cdk-overlay-container"
>
  <div
    class="cdk-overlay-connected-position-bounding-box"
    dir="ltr"
    style="left: 0px; bottom: 0px; height: 0px; width: 0px; align-items: center; justify-content: flex-end;"
  >
    <div
      class="cdk-overlay-pane"
      id="cdk-overlay-0"
      style="pointer-events: auto; position: static; transform: translateY(-5px);"
    >
      <md-tooltip-container>
        <div
          class="md-event-overlay md-event-overlay--top md-popover md-event-overlay--arrow"
          ng-reflect-klass="md-event-overlay"
          ng-reflect-ng-class="[object Object]"
        >
          
          <div
            class="md-event-overlay__arrow"
          />
          <div
            class="md-event-overlay__children"
          >
            <md-tooltip
              ng-reflect-id="md-tooltip-0"
              ng-reflect-max-width="200"
              ng-reflect-text=""
              ng-reflect-tooltip-h-t-m-l=""
            >
              <div
                class="md-tooltip__wrapper"
                id="md-tooltip-0"
                style="max-width: 200px;"
              >
                
                
              </div>
            </md-tooltip>
          </div>
        </div>
      </md-tooltip-container>
    </div>
  </div>
</div>
`;
